netty/src/main/java/io/grpc/transport/netty/NettyClientHandler.java

```diff
@@ -360,15 +358,24 @@ private int getAndIncrementNextStreamId() {
     return id;
   }
 
+  private Http2Stream requireHttp2Stream(int streamId) throws Http2Exception {
+    Http2Stream stream = connection().stream(streamId);
+    if (stream == null) {
```

Comment:
Then can you add a comment saying, "This should be impossible" or similar?

Saying that "Http2Exception is more useful since it will properly shutdown the connection" is a bit concerning if the exception should never happen. The benefit of AssertionError is that if you ever get it, you know there is a bug in the (local) code causing an invariant to be broken. That is equally useful when reading the code.